analyzers/dataframe/MCParticle.cc

```diff
@@ -657,6 +657,143 @@ ROOT::VecOps::RVec<int>  MCParticle::get_indices_ExclusiveDecay::operator() ( RO
+    }
+  }
+
+
+  if (debug) {
+    std::vector<int> unstable_products = get_list_of_particles_from_decay( imother, in, ind ) ;
+    for ( auto & k: unstable_products) {
+      std::cout << " ......... unstable daughter PDG = " << in[k].PDG << std::endl;
+    }
+  }
+
+  std::vector<int> products ;
+  if ( m_stableDaughters ) {
+    products = get_list_of_stable_particles_from_decay( imother, in, ind ) ;
+  }
+  else {
+    products = get_list_of_particles_from_decay( imother, in, ind ) ;
+  }
+
+  if (debug) {
+    for (auto& idx: products ) {
+      std::cout << " ........... decay PDG = " << in[idx].PDG << std::endl;
+    }
+  }
+  std::vector<int> found;
+  for (auto & pdg_d: m_pdg_daughters ) {
+    if (debug) std::cout << " -- looking for PDG = " << pdg_d << std::endl;
+    for (auto & idx_d: products) {
+      if ( (m_chargeConjugateDaughters && abs(in[idx_d].PDG) == abs(pdg_d)) || in[idx_d].PDG == pdg_d) {
+	// careful, there can be several particles with the same PDG !
+	if (std::find(found.begin(), found.end(), idx_d) == found.end())  {  // idx_d has NOT already been "used"
+	  found.push_back( idx_d );
+	  if (debug) std::cout << "       found PDG = " << pdg_d << std::endl;
+	}
+      }
+    }
+  }
+  if ( found.size() >= m_pdg_daughters.size()  && products.size() >= m_pdg_daughters.size()) {  // all daughters have been found. That's the decay mode looked for.
+    result.push_back( imother );
+    for ( auto & idx_d: found) {   // use "found" and not "products", to get the right ordering
+      result.push_back( idx_d );
+    }
+
+    if (debug) {
+      std::cout << " --- found the decay mode requested " << std::endl;
+      for ( auto & id: result ) {
+	std::cout << " idx = " << id << " PDG = " << in[id].PDG << std::endl;
+      }
+    }
+  }
+
+  return result;
+
+}
+
+
+// ----------------------------------------------------------------------------------------------------------------------------------
+
+MCParticle::get_indices_InclusiveDecay::get_indices_InclusiveDecay( int pdg_mother, std::vector<int> pdg_daughters, bool stableDaughters, bool chargeConjugateMother, bool chargeConjugateDaughters) {
+  m_pdg_mother = pdg_mother;
+  m_pdg_daughters = pdg_daughters;
+  m_stableDaughters = stableDaughters;
+  m_chargeConjugateMother = chargeConjugateMother;
+  m_chargeConjugateDaughters = chargeConjugateDaughters;
+} ;
+
+ROOT::VecOps::RVec<int>  MCParticle::get_indices_InclusiveDecay::operator() ( ROOT::VecOps::RVec<edm4hep::MCParticleData> in, ROOT::VecOps::RVec<int> ind) {
+
+  // Look for a specific decay specified by the mother PDG_id and
+  // the PDG_ids of the daughters
+  // Returns a vector with the indicess, in the Particle block, of the mother and of
+  // the daughters - in the order defined by std::vector<int> pdg_daughters.
+  //
+  // In case there are several such decays in the event, keep only the first one.
+
+  ROOT::VecOps::RVec<int>  result;
+
+  //bool debug = true;
+  bool debug = false;
```

Comment:
Could be added to the constructor in order to make it configurable, but I think the debug block can just be removed